### ISSUE TITLE
Script template should have a type requirement

### ISSUE THREAD

**Issue by Jummit** [godotengine/godot#36605]
**Godot version:**

ed0f1940cbdbe470da85f446f315ee12897038a1

**Issue description:**

When creating a new script and selecting a type, it doesn't always match the script template type.

**Steps to reproduce:**

Create a new script and set the type to Object.

**Comment by akien-mga** [comment]
Could you be more specific? What is the script template type?

**Comment by Jummit** [comment]
I mean this dialog:
![20200227](https://user-images.githubusercontent.com/28286961/75482614-7542a800-59a5-11ea-9b16-cfec3e86f115.png)
The default template inherits from "Node", which doesn't match the type I selected, which is "Object".

**Comment by akien-mga** [comment]
I don't understand. I selected "Object" for "Inherits", then clicked "Create", and the script has `extends Object` as expected.

**Comment by dalexeev** [comment]
I think @Jummit meant this:

![200228-1](https://user-images.githubusercontent.com/47700418/75510607-95ea1c80-59fc-11ea-8f81-9884c255aee2.png)

**Comment by akien-mga** [comment]
Thanks... That's definitely information that should have been in the issue from the start, please put more effort in your issues @Jummit, otherwise it just means more work for contributors to figure out what is meant.

**Comment by cat-kitten** [comment]
So when a script extends to "Object" shouldn't there be functions _ready and _process (I'm not that experienced so I don't know)

**Comment by dalexeev** [comment]
Yes. When a script does not inherit the `Node` class, the `_ready` and `_process` functions are not needed in the template.

**Comment by cat-kitten** [comment]
#36678
